colcon_gradle/package_identification/gradle.py

```diff
@@ -29,13 +30,21 @@ def identify(self, metadata):  # noqa: D102
 
         data = extract_data(build_gradle)
         if not data['name'] and not metadata.name:
-            raise RuntimeError(
-                "Failed to extract project name from '%s'" % build_gradle)
+            msg = ("Failed to extract project name from '%s'" % build_gradle)
+            logger.error(msg)
```

Comment:
No need to log it, `RuntimeError` is enough.

Reply:
base on https://github.com/colcon/colcon-core/blob/aca10c55a6c7749777171bb18fce20ea27ca0d1d/colcon_core/package_identification/python.py#L40